src/apirequest.ts

```diff
@@ -228,18 +228,25 @@ async function createAPIRequestAsync<T>(parameters: APIRequestParams) {
 
   if (parameters.mediaUrl && media.body) {
     options.url = parameters.mediaUrl;
-    if (resource) {
+    if (resource || parameters.options.multipart === true) {
```

Comment:
Is the user going to need to know to set this property?  Or is it something we can do under the hood?

Reply:
At the outset the user will need to set `multipart`, if this works well, I would suggest we look into making the generator set this if the API claims to support multi-part uploads.